src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2220,6 +2220,10 @@ void RadioSpinel<InterfaceType, ProcessContextType>::HandleRcpUnexpectedReset(sp
 
     otLogCritPlat("Unexpected RCP reset: %s", spinel_status_to_cstr(aStatus));
 
+#if OPENTHREAD_SPINEL_CONFIG_RCP_CUSTOM_RESTORATION
+    SuccessOrDie(mSpinelInterface.HandleRcpFailed());
```

Comment:
If the custom handler is about resetting the device, I think it should be called in `RecoverFromRcpFailure()`. It's where the real recovery happens.